vim-patch:8.1.{899,903,905,907,908,910,913,1746}

janlazo opened the pull request:
Close https://github.com/neovim/neovim/pull/10051
Close #9930.

Thanks @gafeol.

gafeol committed:
Checks for overflow when parsing string to int

janlazo committed:
vim-patch:8.1.0899: no need to check restricted mode for setwinvar()

Problem:    No need to check restricted mode for setwinvar().
Solution:   Remove check_restricted().
https://github.com/vim/vim/commit/e0fb7d1e38e1cf699412e212cda863420f5f8bd2

janlazo committed:
vim-patch:8.1.0903: struct uses more bytes than needed

Problem:    Struct uses more bytes than needed.
Solution:   Reorder members of regitem_S. (Dominique Pelle, closes vim/vim#3936)
https://github.com/vim/vim/commit/beb7574d6b0eea1cae70aa2913a690da56de5307

janlazo committed:
vim-patch:8.1.0908: can't handle large value for %{nr}v in regexp

Problem:    Can't handle large value for %{nr}v in regexp. (Kuang-che Wu)
Solution:   Give an error if the value is too large. (closes vim/vim#3948)
https://github.com/vim/vim/commit/9403a2168db82b7de80f792984084bb3f00e2263

justinmk commented:
should we use @jamessan 's suggestion in https://github.com/neovim/neovim/pull/10051#issuecomment-500223981 ?

Does this close https://github.com/neovim/neovim/issues/9930 ?

janlazo committed:
regexp: use fixed types to avoid overflow

janlazo commented on src/nvim/regexp_nfa.c:
```diff
@@ -1493,8 +1493,8 @@ static int nfa_regatom(void)
-      long n = 0;
-      int cmp = c;
+      uint64_t n = 0;
+      uint32_t cmp = c;
```
Do I need an assertion for this? `c` is int.

justinmk replied:
> refactoring the other regexp functions to return uint32_t will take a while. 

Just an assert is fine. Refactoring regexp module is out of scope, not worth our time.

janlazo replied:
Actually, does `cmp` need to be fixed? It doesn't impact `n`.

janlazo committed:
vim-patch:8.1.1746: ":dl" is seen as ":dlist" instead of ":delete"

Problem:    ":dl" is seen as ":dlist" instead of ":delete".
Solution:   Do not use cmdidxs2[] if the length is 1. (closes vim/vim#4721)
https://github.com/vim/vim/commit/94f82cbacf76767b5ac32f813e1d670501dbd0e6

janlazo commented on src/nvim/regexp_nfa.c:
```diff
@@ -1493,16 +1493,22 @@ static int nfa_regatom(void)
     {
-      long n = 0;
-      int cmp = c;
+      int64_t n = 0;
```
This has to be a signed type to not trigger the assertion in the new test.

janlazo committed:
vim-patch:8.1.0905: complicated regexp causes a crash

Problem:    Complicated regexp causes a crash. (Kuang-che Wu)
Solution:   Limit the recursiveness of addstate(). (closes vim/vim#3941)
https://github.com/vim/vim/commit/5567ad48b66dff13670af52a48509059acc34dfe

janlazo committed:
vim-patch:8.1.0907: CI tests on AppVeyor are failing

Problem:    CI tests on AppVeyor are failing.
Solution:   Reduce the recursiveness limit for regexp.
https://github.com/vim/vim/commit/5382f12c910b7f8e46acdde5488f26a86f9fcac1

janlazo committed:
vim-patch:8.1.0910: crash with tricky search pattern

Problem:    Crash with tricky search pattern. (Kuang-che Wu)
Solution:   Check for runnning out of memory. (closes vim/vim#3950)
https://github.com/vim/vim/commit/15bbd6ec871a0efdd16256e1fccbaac0fd374cbd

janlazo committed:
vim-patch:8.1.0913: CI crashes when running out of memory

Problem:    CI crashes when running out of memory.
Solution:   Apply 'maxmempattern' also to new regexp engine.
https://github.com/vim/vim/commit/688b3983d8b321e0d32dd51914fa474a0988daf6

janlazo committed:
regexp: add function attributes

justinmk commented on test/functional/eval/match_functions_spec.lua:
```diff
@@ -156,3 +156,12 @@ describe('matchaddpos()', function()
   end)
 end)
+
+describe('nfa_regatom() column search', function()
```
In the merge, I moved this test to a new test file `normal/search_spec.lua`
